### PR TITLE
[DarkModeSetting] cleanup dirty code introduced myself

### PR DESCRIPTION
**Pull request by fm-sys**
If we want to make the code working regardless of the defined enum element names as discussed in https://github.com/stefan-niedermann/nextcloud-deck/pull/755#discussion_r541541697, I just realized that we can't use `valueOf(String)` if we want to do it right. This PR does not cause any functional change but just cleans up this dirty code part.

BTW, should we make the same change for the Notes App, so that there is no longer this magic relation between string resources and enum names? Since I am now familiar with the topic, I could do this quickly...

### CODE REVIEW

**Comment by stefan-niedermann** [comment]
Hmmm 🤔 

Maybe we shouldn't rely on the string values at all but simply store the integer values of the mode IDs like `AppCompatDelegate.MODE_NIGHT_YES`?

This makes it a bit harder to immediately recognize the preference value when looking in the preferences xml, but on the other hand it makes the code much easier.

What do you think?

> BTW, should we make the same change for the Notes App, so that there is no longer this magic relation between string resources and enum names? Since I am now familiar with the topic, I could do this quickly...

Another PR would be awesome! But let's first finish this one 🙂

**Comment by fm-sys** [comment]
I had the same idea at the beginning but had discarded it for some reason. Since the old PR is not yet rolled out, is it enough to edit the old migration script instead of creating a new one?

**Comment by stefan-niedermann** [comment]
> Since the old PR is not yet rolled out, is it enough to edit the old migration script instead of creating a new one?

Yes, absolutely, just edit the old migration script.

We won't release `1.13.2` in the next few days for several reasons (wait for feedback regarding another new feature, wait for transifex, ...)

> discarded it for some reason

If you remember it, just write it down, maybe it actually *has* drawbacks. But at the moment i can not imagine any except that it is not *that* easy to directly read it in the preferences.xml (which not a big deal IMHO).

**Comment by fm-sys** [comment]
> Maybe we shouldn't rely on the string values at all but simply store the integer values of the mode IDs like `AppCompatDelegate.MODE_NIGHT_YES`?

The only problem was, that `ListPreference` needs a `string-array` as `entryValues`. Now the mode IDs are used identifiers and are converted back to int when used in the code...

File: app/src/main/res/values/setup.xml [hunk 1]
```diff
@@ -30,14 +30,17 @@
         <item>@string/pref_value_background_6_hours</item>
     </string-array>
 
-    <string name="pref_value_theme_light" translatable="false">LIGHT</string>
-    <string name="pref_value_theme_dark" translatable="false">DARK</string>
-    <string name="pref_value_theme_system_default" translatable="false">SYSTEM_DEFAULT</string>
+    <!-- @see AppCompatDelegate#MODE_NIGHT_FOLLOW_SYSTEM -->
+    <string name="pref_value_theme_system_default" translatable="false">-1</string>
```
Comment (stefan-niedermann) [inline app/src/main/res/values/setup.xml:34]:
```xml
<integer name="pref_value_theme_system_default">-1</integer>
```

**Comment by stefan-niedermann** [comment]
> `ListPreference` needs a `string-array` as `entryValues`

You are right, pitty!

Though I think we can reference `@integer/xxxxx` values from within the `string-array`, no? And also the `SharedPreferences.Editor` could store the `Integer` directly (via `editor.getInt` and `editor.putInt`)?

File: app/src/main/res/values/setup.xml [hunk 2]
```diff
@@ -30,14 +30,17 @@
...
+    <!-- @see AppCompatDelegate#MODE_NIGHT_NO -->
+    <string name="pref_value_theme_light" translatable="false">1</string>
+    <!-- @see AppCompatDelegate#MODE_NIGHT_YES -->
+    <string name="pref_value_theme_dark" translatable="false">2</string>
 
     <string-array name="darkMode_values">
```
[the remark this answers is not in the document]
Comment (stefan-niedermann) [reply app/src/main/res/values/setup.xml:40]: As you said: We can't use `integer-array`, but i think we can use a `string-array` which references the values with `@integer/xxxx`...

**Comment by fm-sys** [comment]
> Though I think we can reference `@integer/xxxxx` values from within the `string-array`, no?

Damn. You are right! That's why I like contributing to open source projects. You live and learn ;-)

**Comment by fm-sys** [comment]
> And also the `SharedPreferences.Editor` could store the `Integer` directly (via `editor.getInt` and `editor.putInt`)?

This part unfortunately does not work. List preference stores the selected item as a string which cannot be cast to int for retrieval. So IMHO it's the easiest to just stay with string resources...

**Pull request merged**